website/source/intro/getting-started/build-image.html.md

```diff
@@ -200,4 +198,273 @@ image was pretty useless in this case (nothing was changed about it), this page
+
+Your output will look like this: 
+
+```
+amazon-ebs output will be in this color.
+
+==> amazon-ebs: Prevalidating AMI Name: packer-linux-aws-demo-1507231105
+    amazon-ebs: Found Image ID: ami-fce3c696
+==> amazon-ebs: Creating temporary keypair: packer_59d68581-e3e6-eb35-4ae3-c98d55cfa04f
+==> amazon-ebs: Creating temporary security group for this instance: packer_59d68584-cf8a-d0af-ad82-e058593945ea
+==> amazon-ebs: Authorizing access to port 22 on the temporary security group...
+==> amazon-ebs: Launching a source AWS instance...
+==> amazon-ebs: Adding tags to source instance
+    amazon-ebs: Adding tag: "Name": "Packer Builder"
+    amazon-ebs: Instance ID: i-013e8fb2ced4d714c
+==> amazon-ebs: Waiting for instance (i-013e8fb2ced4d714c) to become ready...
+==> amazon-ebs: Waiting for SSH to become available...
+==> amazon-ebs: Connected to SSH!
+==> amazon-ebs: Uploading ./scripts/welcome.txt => /home/ubuntu/
+==> amazon-ebs: Provisioning with shell script: /var/folders/8t/0yb5q0_x6mb2jldqq_vjn3lr0000gn/T/packer-shell661094204
+    amazon-ebs: total 32
+    amazon-ebs: drwxr-xr-x 4 ubuntu ubuntu 4096 Oct  5 19:19 .
+    amazon-ebs: drwxr-xr-x 3 root   root   4096 Oct  5 19:19 ..
+    amazon-ebs: -rw-r--r-- 1 ubuntu ubuntu  220 Apr  9  2014 .bash_logout
+    amazon-ebs: -rw-r--r-- 1 ubuntu ubuntu 3637 Apr  9  2014 .bashrc
+    amazon-ebs: drwx------ 2 ubuntu ubuntu 4096 Oct  5 19:19 .cache
+    amazon-ebs: -rw-r--r-- 1 ubuntu ubuntu  675 Apr  9  2014 .profile
+    amazon-ebs: drwx------ 2 ubuntu ubuntu 4096 Oct  5 19:19 .ssh
+    amazon-ebs: -rw-r--r-- 1 ubuntu ubuntu   18 Oct  5 19:19 welcome.txt
+    amazon-ebs: WELCOME TO PACKER!
+==> amazon-ebs: Provisioning with shell script: ./example.sh
+    amazon-ebs: hello
+==> amazon-ebs: Stopping the source instance...
+    amazon-ebs: Stopping instance, attempt 1
+==> amazon-ebs: Waiting for the instance to stop...
+==> amazon-ebs: Creating the AMI: packer-linux-aws-demo-1507231105
+    amazon-ebs: AMI: ami-f76ea98d
+==> amazon-ebs: Waiting for AMI to become ready...
+```
+
+### A windows example
+
+Note that this uses a larger instance.  You will be charged for it.
+
+You'll need to have a boostrapping file to enable ssh or winrm; here's a basic example of that file.
+
+```
+<powershell>
+
+# set administrator password
+net user Administrator SuperS3cr3t!
+wmic useraccount where "name='Administrator'" set PasswordExpires=FALSE
+
+# First, make sure WinRM doesn't run and can't be connected to
+netsh advfirewall firewall add rule name="WinRM" protocol=TCP dir=in localport=5985 action=block
+net stop winrm
+
+# turn off PowerShell execution policy restrictions
+Set-ExecutionPolicy -ExecutionPolicy Bypass -Scope LocalMachine
+
+# configure WinRM
+winrm quickconfig -q
```

Comment:
@SwampDragons Using `winrm quickconfig -q` here _may_ cause an issue. I realise _a lot_ of the examples out there on the web use this method to configure WinRM. However, I found it _occasionally_ causes Packer to hang immediately after Packer reports the connection over WinRM is made.

The basic order of events leading to the issue is as follows:

1. At the point your bootstrapping file is run Packer will be trying to connect over WinRM with a small delay between each attempt.
2. You initially configure the firewall and stop WinRM to prevent Packer from connecting.
3. The `winrm quickconfig -q` command will set up _everything_ required to allow a connection to be made over WinRM. This _includes_ the firewall. Essentially, after this command is run Packer will be able to connect.
4. Depending on the timing of Packers connection attempt, it's possible that Packer might immediately establish a connection after the `winrm quickconfig -q` command completes.
5. You proceed to configure WinRM.
6. You then _restart_ WinRM to pick up the changes. If Packer has established a connection the restart essentially pulls the rug out from underneath it and we see an issue. Of course if the timings have worked out so that Packer _didn't_ connect, we won't see an issue...

The way to avoid this is to configure WinRM manually. It's a little more involved, but I've been using [this script](https://gist.github.com/DanHam/c1b6f0d3a2d2c46284966e88c48bc9d7) to configure WinRM for my Windows 2012R2 and 2016 instances and have not seen any issues. Hope that helps!

Reply:
Oh, wonderful.  I'll steal that script for docs.  I'm glad you said something; if you have any other windows examples you think would be worth sharing, please let me know -- I'm trying to create some better getting-started docs.😁

Reply:
hmmm actually in order to keep this example as short and non-threatening as possible I think I'm going to keep the config as is -- but I've saved a working copy of this config to use in future examples.

Reply:
@SwampDragons Oh No! Please don't leave that `winrm quickconfig -q` command in.

Like I said it leads to Packer runs failing occasionally so it definitely **shouldn't** be present in any example scripts in the docs! If people see and copy that they will likely then blame Packer for subsequent failures and you'll get bug reports...

I didn't mean to suggest completely replacing your script with my own. I just meant for you to use the bits that manually set up WinRM!

**I've now adapted your script and tested it with Windows 2012R2 and 2016 - It works fine.** It should be OK to use with Windows 2008 onwards. **See the adapted script [HERE](https://gist.github.com/DanHam/78f8d0b310c71f7f80229435c7659ce7). It's only *very slightly* different!**

Note that the script still uses the old style `netsh advfirewall...` commands rather than using the newer PowerShell cmdlets for working with Windows firewall. This means the script should be compatible with Windows 2008 on. 

The preconfigured `"Windows Remote Management (HTTP-In)"` rules are present in all Windows Server editions from 2008 onwards.

Reply:
@DanHam I can't get my example code to work with your gist.  What exactly is necessary to be replacing the quickconfig line?

Reply:
@SwampDragons You used the adaptation of your script [HERE](https://gist.github.com/DanHam/78f8d0b310c71f7f80229435c7659ce7) and that didn't work for you? Does Packer report that you've connected over WinRM? Did you remember to configure the line endings correctly for that script (e.g. run unix2dos on the script)?

It *could be* that UAC is getting in the way. Perhaps try [THIS](https://gist.github.com/DanHam/c5a56346ac1e39589e6df152b4475580) script which includes a further update to your script to sort UAC. I probably should have included that in my first effort at adapting your script.

In answer to your question, the `winrm quickconfig -q` command does the following:

1. Sets the WinRM service start up type to 'Automatic'
2. Creates a new HTTP listener
3. Configures Windows Firewall to allow management messages to be sent over HTTP (port 5985).
4. Configures UAC to allow privilege elevation in remote shells. This is the change to the `LocalAccountTokenFilterPolicy` registry setting that you can now see in the [updated script](https://gist.github.com/DanHam/c5a56346ac1e39589e6df152b4475580).

As you can see you were already doing most of those things in your script. To replace the `winrm quickconfig` command we needed to add commands for steps 2 and 4 and obviously retain the parts of the script that were doing steps 1 and 3 for us.

The problem with using the `winrm quickconfig -q` command is that step 3 will undo anything you previously did with the firewall to prevent Packer making a connection. In other words `winrm quickconfig -q` is a little *too efficient* and allows Packer to make the connection over WinRM before you are actually intending e.g. when you manually update the firewall to allow the connection at the end of your script.

Let me know how you get on. I've tested the [new script](https://gist.github.com/DanHam/c5a56346ac1e39589e6df152b4475580) again with my Windows 2012R2 and 2016 builds (Virtualbox and VMware ISO builds) and it works fine. The script should configure WinRM correctly for Windows 2008 and later.

Reply:
This is super weird.  When I use either of your bootstrap scripts with my AWS example in this document, it hangs waiting for WinRM to connect.  Never connects.

Reply:
@SwampDragons Dammit! I'm really sorry. I've now worked out why you were having problems. There's a small difference between the firewall command you need to run against an Amazon source AMI and an instance built from an ISO. 

I've now made 100% sure everything works by testing with [these templates and scripts](https://github.com/DanHam/packer-testing/tree/master/aws/windows) using the Amazon EBS builder.

Since you already merged this I've submitted a fresh PR with the changes and a few additional fixes to things I saw along the way. Hopefully, you will be able to quickly verify the changes by using the templates and scripts referenced above.

Apologies again for any time wasted trying to get my first efforts working...

Reply:
No worries!  Thanks for being so dedicated to getting this right.